### PR TITLE
Item Pickups and Inventory Interactions

### PR DESCRIPTION
**Pull request by kylegessel**
This pull request fortifies our earlier implementation of item inventory and cleans up the player pickup class by giving each item their own pickup class.

Final remaining items are also added. Bomb, boomerang, and Blue Rupees have been added to the entities list and can now be found as items in the first room.

Link now responds to changes in his inventory and checks his inventory before executing commands.
- Link no longer uses the thrown sword if he has less than full health.
- Link no longer uses a bomb if has none remaining in his inventory.
- Link can only use the boomerang after he has picked it up for the first time.

There was also some tweaking of the sword hit-box. It was not being used previously, but now can be used to kill enemies. Its class still needs some attention and its one of the classes I really want to rework.

HUD items (compass and map) as well as the clock still need method bodies.

### CODE REVIEW

**Comment by kylegessel** [comment]
Newest push added new items to the dictionary, and cleaned up some the the pickup classes.

Boomerang can now only be found as a pickup in room 11.

Arrows won't fire unless the link has picked up the bow!

File: Sprintfinity3902/Entities/Items/SwordHitboxItem.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public override void Update(GameTime gameTime)
 
         public void MoveItem()
         {
-            if (itemUseCount <= 5) //this time amount needs to be tweaked.
+            if (itemUseCount <= 30) //this time amount needs to be tweaked.
```
Comment (Bailey-Ardrey) [inline Sprintfinity3902/Entities/Items/SwordHitboxItem.cs]: At some point this Magic Number might need to be taken out.

File: Sprintfinity3902/States/ItemPickup/FairyPickup.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        { 
+
+        }
+
+        public bool Pickup(Player Link)
+        {
+            Link.LinkHealth = Link.MaxHealth;
+            Link.heartChanged = true;
+
+            Sound.SoundLoader.Instance.GetSound(Sound.SoundLoader.Sounds.LOZ_Get_Heart).Play(Global.Var.VOLUME, Global.Var.PITCH, Global.Var.PAN);
```
Comment (Bailey-Ardrey) [inline Sprintfinity3902/States/ItemPickup/FairyPickup.cs]: I am not sure if the Fairy has a specific sound associated.

File: Sprintfinity3902/States/ItemPickup/HeartContainerPickup.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        { 
+
+        }
+
+        public bool Pickup(Player Link)
+        {
+            Link.MaxHealth += 2;
+            Link.LinkHealth += 2;
+            Link.heartChanged = true;
+            Sound.SoundLoader.Instance.GetSound(Sound.SoundLoader.Sounds.LOZ_Get_Heart).Play(Global.Var.VOLUME, Global.Var.PITCH, Global.Var.PAN);
```
Comment (Bailey-Ardrey) [inline Sprintfinity3902/States/ItemPickup/HeartContainerPickup.cs]: I am not sure if the Heart Container has a specific sound associated.

File: Sprintfinity3902/States/ItemPickup/HeartPickup.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            if (Link.LinkHealth < Link.MaxHealth)
+            {
+                Link.LinkHealth++;
+                if (Link.LinkHealth != Link.MaxHealth)
+                {
+                    Link.LinkHealth++;
+                }
+                Link.heartChanged = true;
+                Sound.SoundLoader.Instance.GetSound(Sound.SoundLoader.Sounds.LOZ_Get_Heart).Play(Global.Var.VOLUME, Global.Var.PITCH, Global.Var.PAN);
```
Comment (Bailey-Ardrey) [inline Sprintfinity3902/States/ItemPickup/HeartPickup.cs]: Heart pickup sound is not working at runtime for me. Are you having the same issue?

**Review by Bailey-Ardrey — CHANGES_REQUESTED** [review]
Everything looks good really good! Pickup cleans up a lot in player. Only two things I noticed:

1. The heart sound isn't playing when Link picks up a heart.
2. An error is thrown when Link dies in this iteration of the code... Game Over state is broken.

**Comment by Bailey-Ardrey** [comment]
It actually looks like Game Over was also broken in my pr too.

**Comment by Bailey-Ardrey** [comment]
It has to do with the Sound Manager being Reset in the Win and Lose Wrapper. I think we can ignore it for now, as it was broken before this pr.

File: Sprintfinity3902/States/ItemPickup/HeartPickup.cs (hunk 4 above)
Comment (kylegessel) [reply Sprintfinity3902/States/ItemPickup/HeartPickup.cs]: Yup, this is a bug. Heart sound only plays if link is damaged.

Updating so it plays on every pickup.

File: Sprintfinity3902/States/ItemPickup/FairyPickup.cs (hunk 2 above)
Comment (kylegessel) [reply Sprintfinity3902/States/ItemPickup/FairyPickup.cs]: Removing it now!

File: Sprintfinity3902/States/ItemPickup/HeartContainerPickup.cs (hunk 3 above)
Comment (kylegessel) [reply Sprintfinity3902/States/ItemPickup/HeartContainerPickup.cs]: Removed it!

File: Sprintfinity3902/Entities/Items/SwordHitboxItem.cs (hunk 1 above)
Comment (kylegessel) [reply Sprintfinity3902/Entities/Items/SwordHitboxItem.cs]: Thanks for catching this, just gave it a constant. I still want to take a look at this class/number in the future but for now its changed.

**Comment by kylegessel** [comment]
Added the changes requested!
 
- Some items had their sound removed (Fairy and Heart Container). 
- Heart sound plays when link picks up a heart, even while full health.
- Removed a magic number.

**Review by Bailey-Ardrey — APPROVED** [review]
Looks good! Ready to merge. I'll let Casey get his comments in first.

**Comment by caseyvanc** [comment]
This looks good to me. I like the item pickup implementation.

**Pull request merged**